funcx_endpoint/funcx_endpoint/endpoint/interchange.py

```diff
@@ -282,6 +286,13 @@ def start(self):
 
             try:
                 self._start_threads_and_main()
+            except EndpointLockedError:
+                # We want to stop the retry loop on some specific exception types
+                log.warning("Endpoint is currently locked.  Exiting.")
+                self._kill_event.set()
+            except EndpointInUseError:
+                log.warning("Endpoint is already active elsewhere.  Exiting.")
+                self._kill_event.set()
```

Comment:
Probably worth it to leave a comment for future devs pointing out how these errors are expected to be raised, so they don't have to trace out the whole `_start_threads_and_main` flow

Reply:
Added.